libraries/chain/include/eosio/chain/merkle.hpp

```diff
@@ -9,6 +11,18 @@ namespace eosio::chain {
 
 namespace detail {
 
+struct merkle_pool {
+   merkle_pool() {
+      thread_pool.start(4, [](const fc::exception&){}, [](size_t){});
```

Comment:
idk what to do with exceptions off hand here. Need to think about it

Reply:
Log the exception and `std::exit(-2);`. Exceptions should be completely unexpected. If they can happen then they should be caught and handled as a bad hash in `calculate_merkle_pow2`.

Reply:
I had to check it worked like I would have guess -- and yeah, when using `use_future` exceptions thrown inside of the dispatched work do get packaged in to the future. So if you're only post()ing `use_future`s then it is completely unexpected for anything to leak out.

It might be useful for `named_thread_pool` to provide some pre-defined functions one can pick up and use. `thread_pool.start(4, named_thread_pool::abort_on_exception,...)` or something. Will just do something locally here for now. I'm not sure `exit()` is the right choice since it seems like you'd want something more aborty

Reply:
`std::abort()` does make sense here.
It does seem having a `named_thread_pool::abort_on_exception` would be handy.